agent/engine/docker_task_engine_test.go

```diff
@@ -1380,6 +1380,89 @@ func TestPullNormalImage(t *testing.T) {
 	assert.Equal(t, dockerapi.DockerContainerMetadata{}, metadata, "expected empty metadata")
 }
 
+func TestPullImageWithNeverAgentPullBehavior(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, _, _, privateTaskEngine, _, imageManager, _ := mocks(t, ctx, &config.Config{AgentPullBehavior: config.NeverAgentPullBehavior})
+	defer ctrl.Finish()
+	taskEngine, _ := privateTaskEngine.(*DockerTaskEngine)
+	saver := mock_statemanager.NewMockStateManager(ctrl)
+	taskEngine.SetSaver(saver)
+	taskEngine._time = nil
+	imageName := "image"
+	container := &api.Container{
+		Type:  api.ContainerNormal,
+		Image: imageName,
+	}
+	task := &api.Task{
+		Containers: []*api.Container{container},
+	}
+	imageState := &image.ImageState{
+		Image: &image.Image{ImageID: "id"},
+	}
+
+	imageManager.EXPECT().RecordContainerReference(container)
+	imageManager.EXPECT().GetImageStateFromImageName(imageName).Return(imageState)
+	saver.EXPECT().Save()
+	metadata := taskEngine.pullContainer(task, container)
+	assert.Equal(t, dockerapi.DockerContainerMetadata{}, metadata, "expected empty metadata")
+}
+
+func TestPullImageWithOnceAgentPullBehavior(t *testing.T) {
```

Comment:
Can you also add a test where config is `once` and `PullSucceeded` is false?

Reply:
OK